rest-backend/src/main/resources/db/update_knowledge_db.sql

```diff
@@ -0,0 +1,432 @@
+DROP CONSTRAINT fk_49ig9c71ttnmsowf1v6yx36v0, -- @parameter
+    ADD CONSTRAINT fk_49ig9c71ttnmsowf1v6yx36v0 -- @parameter
+    FOREIGN KEY (bug_construct_change)
+    REFERENCES bug_construct_change (id)
+    ON UPDATE CASCADE
+    ON DELETE NO ACTION;
+	
+ALTER TABLE bug_affected_construct_change
+DROP CONSTRAINT fk_7dap5l5oskn621gbnfh5fbhc9, -- @parameter
+    ADD CONSTRAINT fk_7dap5l5oskn621gbnfh5fbhc9 -- @parameter
+    FOREIGN KEY (affected_lib)
+    REFERENCES bug_affected_library (id)
+    ON UPDATE CASCADE
+    ON DELETE NO ACTION;
+
+-- Create temporary tables for the data
+
+CREATE TABLE new_bug (like bug including all);
+CREATE TABLE new_bug_affected_library (like bug_affected_library including all);
+CREATE TABLE new_bug_construct_change (like bug_construct_change including all);
+CREATE TABLE new_bug_references (like bug_references including all);
+CREATE TABLE new_construct_id (like construct_id including all);
+CREATE TABLE new_library_id (like library_id including all);
+
+-- Copy data from files into temporary tables
+-- Currently relative path used - absolute also possible
+
+\copy new_bug FROM 'bug.sql'; -- @parameter
+--COPY new_bug FROM 'bug.sql';
+
+\copy new_bug_affected_library FROM 'bug_affected_library.sql'; -- @parameter
+--COPY new_bug_affected_library FROM 'bug_affected_library.sql'
+
+\copy new_bug_construct_change FROM 'bug_construct_change.sql'; -- @parameter
+--COPY new_bug_construct_change FROM 'bug_construct_change.sql'
+
+\copy new_bug_references FROM 'bug_references.sql'; -- @parameter
+--COPY new_bug_references FROM 'bug_references.sql'
+
+\copy new_construct_id FROM 'construct_id.sql'; -- @parameter
+--COPY new_construct_id FROM 'construct_id.sql'
+
+\copy new_library_id FROM 'library_id.sql'; -- @parameter
+--COPY new_library_id FROM 'library_id.sql'
+
+-- Should return 0
+SELECT loop_update_bug_table_ids();
+
+-- Insert new bugs
+INSERT INTO bug SELECT new_bug.* FROM new_bug, (SELECT new_bug.id FROM new_bug EXCEPT ALL SELECT bug.id FROM bug) AS tmp_bug
+	WHERE new_bug.id = tmp_bug.id;
+
+-- Should return 0
+SELECT loop_update_library_id_table();
+
+-- Insert new library_ids
+INSERT INTO library_id SELECT new_library_id.* FROM new_library_id, (SELECT new_library_id.id FROM new_library_id EXCEPT ALL SELECT library_id.id FROM library_id) AS tmp_library_id
+	WHERE new_library_id.id = tmp_library_id.id;
+
+-- Should return 0
+SELECT loop_update_construct_id_table();
+
+-- Insert new construct_ids
+INSERT INTO construct_id SELECT new_construct_id.* FROM new_construct_id, (SELECT new_construct_id.id FROM new_construct_id EXCEPT ALL SELECT construct_id.id FROM construct_id) AS tmp_construct_id
+	WHERE new_construct_id.id = tmp_construct_id.id;
+
+-- bug_affected_library is empty, insert all as is
+INSERT INTO bug_affected_library SELECT * FROM new_bug_affected_library;
+
+-- Create temporary bug_affected_construct_change with serial id
+
+CREATE TABLE public.tmp_bug_construct_change
+(
+    id SERIAL NOT NULL,
```

Comment:
Does this mean that the ids of the final table bug_construct_change will always be integers from 1 to the number of rows inserted by the script (old + new)? This also implies that the same ID number may be used in multiple places whereas the IDs as handled by hibernate are unique (assigned from the hibernate_sequence). I am not sure whether this can pose issues for JPA.

Reply:
The requested changes were added.